core/tee/tee_fs_key_manager.c

```diff
@@ -85,6 +92,15 @@ static TEE_Result fek_crypt(TEE_OperationMode mode,
 	if (tee_fs_ssk.is_init == 0)
 		return TEE_ERROR_GENERIC;
 
+	res = tee_ta_get_current_session(&sess);
+	if (res != TEE_SUCCESS)
+		return res;
+
+	res = do_hmac(tsk, sizeof(tsk), tee_fs_ssk.key, TEE_FS_KM_SSK_SIZE,
+		      (uint8_t *)&sess->ctx->uuid, sizeof(TEE_UUID));
```

Comment:
This does the hmac on the non-decoded uuid. This is probably fine, since I don't think we are likely to be worried about moving the TA stored data to a different endianness.